File: gaia-sdk-python/gaia_sdk/graphql/request/type/DeletedRoleImpulse.py

from typing import Callable

from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.type.RoleKeyOne import RoleKeyOne


class DeletedRoleImpulse(list):
    """
    Impulse which indicates the result of a delete role impulse
    """

    def id(self):
        self.append(lambda x: "id")

    def data(self, config: Callable[['RoleKeyOne'], None]):
        def callback(registry: VariableRegistry):
            entity = RoleKeyOne()
            config(entity)
            return "data {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/DeletedRoleImpulse.ts

import {RoleKeyOne} from "./RoleKeyOne";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a delete role impulse
 */
export class DeletedRoleImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "DeletedRoleImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:RoleKeyOne) => void) => this.push((registry) => {
        const entity = new RoleKeyOne();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

